api/routes/project.js

```diff
@@ -16,17 +16,17 @@ router.get('/', function(req, res) {
 router.get('/:id', function(req,res) {
   Project.findOne({_id: req.params.id}, function(err, project) {
     if (err || !project) {
-      res.sendStatus('Error in saving project: ' + err);
+      res.send('Error in saving project: ' + err);
     }
     res.json(project);
   })
 })
 
 // update project by id
-router.post('/update/:id', function(req,res) {
-  Project.findOne({_id: req.params.id}, function(err, project) {
+router.post('/update', function(req,res) {
+  Project.findOneAndUpdate({_id: req.query.id}, {[req.query.updateKey]: req.query.updateObject, modifiedAt: new Date()}, function(err, project) {
```

Comment:
It looks like on line 28 and 31, you want to access the project that was updated, but there is no way to access it. I think you're missing a callback.